sdk/core/Azure.Core/src/Shared/MutableJsonElement.WriteTo.cs

```diff
@@ -109,5 +118,95 @@ private void WriteArray(string path, int highWaterMark, JsonElement element, Utf
 
             writer.WriteEndArray();
         }
+
+        internal void WritePatch(Utf8JsonWriter writer)
+        {
+            if (!_root.Changes.HasChanges)
+            {
+                return;
+            }
+
+            // This version is not efficient; Proof of concept.
+
+            // TODO: rewrite using Span, so much win there
+            IEnumerable<string> changed = _root.Changes.GetChangedProperties();
+
+            string jsonPath = string.Empty;
+
+            MutableJsonElement current = _root.RootElement;
+
+            // TODO: this breaks for arrays
+            writer.WriteStartObject();
+
+            foreach (string path in changed)
+            {
+                string currentPath = string.Empty;
+                string[] segments = path.Split(MutableJsonDocument.ChangeTracker.Delimiter);
+                string name = segments[0];
+                currentPath = MutableJsonDocument.ChangeTracker.PushProperty(currentPath, name);
+
+                while (!currentPath.StartsWith(jsonPath))
```

Comment:
This does culture aware comparison. I wonder if ordinal or at least invariant comparison would not work.

Reply:
With the move to Spans, I believe this calls through to SequenceEqual now, which should be culturally invariant, but please let me know if this is incorrect.